File: mobile/app/src/main/java/com/praeter/ui/mainactivity/fragment/classes/ClassesFragment.java
package com.praeter.ui.mainactivity.fragment.classes;

import android.annotation.SuppressLint;
import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.praeter.R;
import com.praeter.core.utils.PraeterNetworkManager;
import com.praeter.data.local.model.Class;

import java.util.ArrayList;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.Unbinder;
import timber.log.Timber;


@SuppressLint("NonConstantResourceId")
public class ClassesFragment extends Fragment
        implements ClassClickListener {

    private Context context;

    @BindView(R.id.rv_classes)
    RecyclerView rvClasses;
    private Unbinder unbinder;

    /**
     * passing data between fragments
     */
    private OnClassItemSelectedListener listener;

    public interface OnClassItemSelectedListener {
        void onClassClick(String className, String classType);
    }


    public static ClassesFragment newInstance() {

        Bundle args = new Bundle();

        ClassesFragment fragment = new ClassesFragment();
        fragment.setArguments(args);
        return fragment;
    }


    ///////////////////////////////////////////
    //
    //  Override methods
    //
    ///////////////////////////////////////////
    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnClassItemSelectedListener) {
            listener = (OnClassItemSelectedListener) context;
        } else {
            throw new ClassCastException(context.toString()
                    + " must implement MainActivity.OnClassItemSelectedListener");
        }
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {

        View root = inflater.inflate(R.layout.fragment_classes, container, false);

        unbinder = ButterKnife.bind(this, root);

        return root;
    }

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);

        context = getActivity();

        // TODO : check internet connection
        if (!PraeterNetworkManager.isConnected(context)) {
            String errorMessage = "Not connected to the internet. Check your internet connection.";
            Timber.e(errorMessage);
            Toast.makeText(context, errorMessage, Toast.LENGTH_LONG).show();
        } else {


            List<Class> classList = prepareClassData();

            ClassAdapter adapter = new ClassAdapter(context, classList, this);

            LinearLayoutManager linearLayoutManager = new LinearLayoutManager(context);
            rvClasses.setLayoutManager(linearLayoutManager);
            rvClasses.setItemAnimator(new DefaultItemAnimator());
            rvClasses.setAdapter(adapter);
        }
    }

    private List<Class> prepareClassData() {
        List<Class> list = new ArrayList<>();

        list.add(new Class("Cours 1 ", "Dance Class"));
        list.add(new Class("Cours 2 ", "Art Class"));
        list.add(new Class("Cours 3 ", "Gastronomic Class"));
        list.add(new Class("Cours 4 ", "Gastronomic Class"));
        list.add(new Class("Cours 5 ", "Dance Class"));
        list.add(new Class("Cours 6 ", "Art Class"));
        list.add(new Class("Cours 7 ", "Art Class"));

        return list;
    }

    @Override
    public void onDestroyView() {
        if (null != unbinder)
            unbinder.unbind();
        super.onDestroyView();
    }

    @Override
    public void onClassItemCLickListener(View view, Class item, int position) {

        // send data to activity
        listener.onClassClick(item.getName(), item.getType());
    }
}

File: mobile/app/src/main/java/com/praeter/ui/mainactivity/fragment/classes/ClassesFragment.kt
package com.praeter.ui.mainactivity.fragment.classes

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import com.praeter.core.utils.PraeterNetworkManager
import com.praeter.data.local.model.ClassModel
import com.praeter.databinding.FragmentClassesBinding
import com.praeter.ui.base.BaseFragment
import timber.log.Timber
import java.util.*


class ClassesFragment : BaseFragment(), ClassClickListener {

    private var _viewBinding: FragmentClassesBinding? = null

    // This property is only valid between onCreateView and
    // onDestroyView.
    private val binding get() = _viewBinding!!

    /**
     * passing data between fragments
     */
    private var listener: OnClassItemSelectedListener? = null

    ///////////////////////////////////////////
    //
    //  Override methods
    //
    ///////////////////////////////////////////
    override fun onAttach(context: Context) {
        super.onAttach(context)
        listener = if (context is OnClassItemSelectedListener) {
            context
        } else {
            throw ClassCastException(
                context.toString()
                        + " must implement MainActivity.OnClassItemSelectedListener"
            )
        }
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _viewBinding = FragmentClassesBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)

        // TODO : check internet connection
        if (!PraeterNetworkManager.isConnected(requireContext())) {
            val errorMessage = "Not connected to the internet. Check your internet connection."
            Timber.e(errorMessage)
            Toast.makeText(context, errorMessage, Toast.LENGTH_LONG).show()
        } else {
            val classModelList: List<ClassModel> = prepareClassData()
            val adapter = ClassAdapter(
                requireContext(), classModelList, this
            )
            val linearLayoutManager = LinearLayoutManager(context)
            binding.rvClasses.layoutManager = linearLayoutManager
            binding.rvClasses.itemAnimator = DefaultItemAnimator()
            binding.rvClasses.adapter = adapter
        }
    }

    private fun prepareClassData(): List<ClassModel> {
        val list: MutableList<ClassModel> = ArrayList<ClassModel>()
        list.add(ClassModel("Cours 1 ", "Dance Class"))
        list.add(ClassModel("Cours 2 ", "Art Class"))
        list.add(ClassModel("Cours 3 ", "Gastronomic Class"))
        list.add(ClassModel("Cours 4 ", "Gastronomic Class"))
        list.add(ClassModel("Cours 5 ", "Dance Class"))
        list.add(ClassModel("Cours 6 ", "Art Class"))
        list.add(ClassModel("Cours 7 ", "Art Class"))
        return list
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _viewBinding = null
    }

    override fun onClassItemCLickListener(view: View?, item: ClassModel, position: Int) {

        // send data to activity
        listener!!.onClassClick(item.name, item.type)
    }

    interface OnClassItemSelectedListener {
        fun onClassClick(className: String?, classType: String?)
    }

    companion object {
        fun newInstance(): ClassesFragment {
            val args = Bundle()
            val fragment = ClassesFragment()
            fragment.arguments = args
            return fragment
        }
    }
}